src/Atom.cpp

```diff
@@ -27,55 +27,59 @@ static bool is_element(const std::string& name) {
     "Th", "Pa", "U" , "Np", "Pu", "Am", "Cm", "Bk", "Cf", "Es", "Fm", "Md", "No", "Lr"
     };
     // clang-format on
-    for (auto& element : ALL_ELEMENTS) {
-        if (name == element) {
+    for (auto& each_element : ALL_ELEMENTS) {
+        if (element == each_element) {
             return true;
         }
     }
     return false;
+
 }
 
-Atom::Atom(std::string name) : name_(std::move(name)), mass_(0), charge_(0) {
-    if (is_element(name_)) {
+Atom::Atom(std::string element, std::string label)
+	: element_(std::move(element)), label_(std::move(label)), mass_(0),
```

Comment:
Instead of the conditional, I added a new `Atom` constructor method that only takes an "element" argument and assigns it to `element_` and `label_`, so the decision is made at compile time. Is this OK, or waaaay too much trouble for such a simple task?